metricbeat/helper/conversion.go

```diff
@@ -1,85 +1,168 @@
-/*
-The conversion functions take a key and a map[string]string. First it checks if the key exists and logs and error
-if this is not the case. Second the conversion to the type is done. In case of an error and error is logged and the
-default values is returned. This guarantees that also if a field is missing or is not defined, still the full metricset
-is returned.
-*/
 package helper
 
 import (
+	"fmt"
 	"strconv"
 
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 )
 
+// Schema describes how a map[string]string object can be parsed and converted into
+// an event. The conversions can be described using an (optionally nested) common.MapStr
+// that contains Conv objects.
+type Schema struct {
+	conversions common.MapStr
```

Comment:
+1 on this, especially for type safeness. I'll try it in #2040 or in a cleanup PR after that. The Mongodb code complicates all of this, because it's input is `map[string]interface{}` not `map[string]string`. It would be nice to find a solution that doesn't involve to much repetition of these types.